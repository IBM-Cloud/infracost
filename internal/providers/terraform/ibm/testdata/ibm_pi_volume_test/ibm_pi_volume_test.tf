terraform {
  required_providers {
    ibm = {
      source = "IBM-Cloud/ibm"
    }
  }
}

provider "ibm" {
  region = "us-south"
}

resource "ibm_resource_group" "resource_group" {
  name = "default"
}

resource "ibm_resource_instance" "powervs_service" {
  name              = "Power instance"
  service           = "power-iaas"
  plan              = "power-virtual-server-group"
  location          = "us-south"
  resource_group_id = ibm_resource_group.resource_group.id
}

resource "ibm_pi_volume" "pi_volume_affinity_set" {
  pi_volume_name       = "affinity-volume-set"
  pi_cloud_instance_id = ibm_resource_instance.powervs_service.guid
  pi_volume_size       = 100
  pi_affinity_policy   = "affinity"
  pi_volume_shareable  = false
}

resource "ibm_pi_volume" "pi_volume_volume_set" {
  pi_volume_name       = "volume-pool"
  pi_cloud_instance_id = ibm_resource_instance.powervs_service.guid
  pi_volume_size       = 100
  pi_volume_pool       = "volume-pool-name"
  pi_volume_shareable  = false
}

resource "ibm_pi_volume" "pi_volume_tier0" {
  pi_volume_name       = "tier0-volume"
  pi_cloud_instance_id = ibm_resource_instance.powervs_service.guid
  pi_volume_size       = 100
  pi_volume_type       = "tier0"
  pi_volume_shareable  = false
}

resource "ibm_pi_volume" "pi_volume_tier1" {
  pi_volume_name       = "tier1-volume"
  pi_cloud_instance_id = ibm_resource_instance.powervs_service.guid
  pi_volume_size       = 100
  pi_volume_type       = "tier1"
  pi_volume_shareable  = false
}

resource "ibm_pi_volume" "pi_volume_tier3" {
  pi_volume_name       = "tier3-volume"
  pi_cloud_instance_id = ibm_resource_instance.powervs_service.guid
  pi_volume_size       = 100
  pi_volume_type       = "tier3"
  pi_volume_shareable  = false
}

resource "ibm_pi_volume" "pi_volume_tier5k" {
  pi_volume_name       = "tier5k-volume"
  pi_cloud_instance_id = ibm_resource_instance.powervs_service.guid
  pi_volume_size       = 100
  pi_volume_type       = "tier5k"
  pi_volume_shareable  = false
}

resource "ibm_pi_volume" "pi_volume_tier0_no_usage" {
  pi_volume_name       = "tier0-volume2"
  pi_cloud_instance_id = ibm_resource_instance.powervs_service.guid
  pi_volume_size       = 100
  pi_volume_type       = "tier0"
  pi_volume_shareable  = false
}

resource "ibm_pi_volume" "pi_volume_tier1_no_usage" {
  pi_volume_name       = "tier1-volume2"
  pi_cloud_instance_id = ibm_resource_instance.powervs_service.guid
  pi_volume_size       = 100
  pi_volume_type       = "tier1"
  pi_volume_shareable  = false
}

resource "ibm_pi_volume" "pi_volume_tier3_no_usage" {
  pi_volume_name       = "tier3-volume2"
  pi_cloud_instance_id = ibm_resource_instance.powervs_service.guid
  pi_volume_size       = 100
  pi_volume_type       = "tier3"
  pi_volume_shareable  = false
}

resource "ibm_pi_volume" "pi_volume_tier5k_no_usage" {
  pi_volume_name       = "tier5k-volume2"
  pi_cloud_instance_id = ibm_resource_instance.powervs_service.guid
  pi_volume_size       = 100
  pi_volume_type       = "tier5k"
  pi_volume_shareable  = false
}
