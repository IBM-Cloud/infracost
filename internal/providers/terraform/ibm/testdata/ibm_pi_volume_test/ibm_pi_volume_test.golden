
 Name                                                                                      Monthly Qty  Unit                                Monthly Cost 
                                                                                                                                                         
 ibm_pi_volume.pi_volume_affinity_set                                                                                                                    
 └─ Volume (100 GB, affinity-volume-set) Price dependent on affinity settings                   73,000  Gigabyte Instance Hours                    $0.00 
                                                                                                                                                         
 ibm_pi_volume.pi_volume_tier0                                                                                                                           
 └─ Volume (100 GB, tier0, tier0-volume)                                                        73,000  Gigabyte Instance Hours                   $24.02 
                                                                                                                                                         
 ibm_pi_volume.pi_volume_tier0_no_usage                                                                                                                  
 └─ Volume (100 GB, tier0, tier0-volume2)                                      Monthly cost depends on usage: $0.000329 per Gigabyte Instance Hours      
                                                                                                                                                         
 ibm_pi_volume.pi_volume_tier1                                                                                                                           
 └─ Volume (100 GB, tier1, tier1-volume)                                                        73,000  Gigabyte Instance Hours                   $20.00 
                                                                                                                                                         
 ibm_pi_volume.pi_volume_tier1_no_usage                                                                                                                  
 └─ Volume (100 GB, tier1, tier1-volume2)                                      Monthly cost depends on usage: $0.000274 per Gigabyte Instance Hours      
                                                                                                                                                         
 ibm_pi_volume.pi_volume_tier3                                                                                                                           
 └─ Volume (100 GB, tier3, tier3-volume)                                                        73,000  Gigabyte Instance Hours                   $10.45 
                                                                                                                                                         
 ibm_pi_volume.pi_volume_tier3_no_usage                                                                                                                  
 └─ Volume (100 GB, tier3, tier3-volume2)                                      Monthly cost depends on usage: $0.000143165 per Gigabyte Instance Hours   
                                                                                                                                                         
 ibm_pi_volume.pi_volume_tier5k                                                                                                                          
 └─ Volume (100 GB, tier5k, tier5k-volume)                                                      73,000  Gigabyte Instance Hours                   $28.84 
                                                                                                                                                         
 ibm_pi_volume.pi_volume_tier5k_no_usage                                                                                                                 
 └─ Volume (100 GB, tier5k, tier5k-volume2)                                    Monthly cost depends on usage: $0.000395 per Gigabyte Instance Hours      
                                                                                                                                                         
 ibm_pi_volume.pi_volume_volume_set                                                                                                                      
 └─ Volume pool (100 GB, volume-pool) Price dependent on volume pool settings                   73,000  Gigabyte Instance Hours                    $0.00 
                                                                                                                                                         
 ibm_resource_instance.powervs_service                                                                                                                   
 └─ Workspace for Power Virtual Server                                                               1  Workspace                                  $0.00 
                                                                                                                                                         
 OVERALL TOTAL                                                                                                                                    $83.31 
──────────────────────────────────
12 cloud resources were detected:
∙ 11 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file
∙ 1 was free:
  ∙ 1 x ibm_resource_group